File: legacy_python/test_summary.py
import asyncio
import os
import json
import logging
from notebooklm_client import NotebookLMClient

logging.basicConfig(level=logging.INFO)

async def main():
    print("Initializing NotebookLM Client...")
    # headless=False to allow manual login if needed, though client logic should handle existing tokens
    client = NotebookLMClient(headless=False)
    
    try:
        await client.start()
        
        video_url = "https://www.youtube.com/watch?v=Kg2Ux47hwg4"
        print(f"Getting summary for: {video_url}")
        
        summary = await client.get_summary_for_video(video_url)
        
        print("\n=== SUMMARY ===\n")
        print(summary)
        print("\n===============\n")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        # await client.stop()
        pass

if __name__ == "__main__":
    asyncio.run(main())

File: src/test_summary.ts

import { NotebookLMClient } from "./notebooklm_client.js";

async function main() {
    const videoUrl = "https://www.youtube.com/watch?v=Kg2Ux47hwg4"; // Test video
    console.log(`Testing Summary Generation for: ${videoUrl}`);

    const client = new NotebookLMClient(true); // Headless
    try {
        await client.start();
        const summary = await client.generateSummary(videoUrl);
        console.log("\n=== FINAL SUMMARY ===\n");
        console.log(summary);
    } catch (e: any) {
        console.error("Test Failed:", e);
        if (e.message.includes("Authentication required")) {
            console.log("\n⚠️  You need to run 'npm run setup' to log in first!");
        }
    } finally {
        await client.stop();
    }
}

main();
